ios/Capacitor/Capacitor/Plugins/LocalNotifications.swift

```diff
@@ -338,6 +338,9 @@ public class CAPLocalNotificationsPlugin: CAPPlugin {
 
         createdCategories.append(generalCategory)
         for type in actionTypes {
+            guard let type = type as? JSObject else {
+              continue
+            }
```

Comment:
Can something like this be used in this 3 fors instead of the guards?

```suggestion
        for case let type as JSObject in actionTypes
```

Reply:
That would work. I am little concerned, though, that that kind of syntax might be too esoteric just to save a couple of lines. It could be hard for someone less familiar with Swift to understand if they looked at the core plugins for examples of how to write a plugin.

Reply:
I didn't propose it for saving lines, but because I find the guard with continue confusing.

Could be possible to indicate in some way that the whole JSArray contain elements of the same type so don't need to cast them inside the for?

In the change I proposed (`guard let types = call.getArray("types", JSObject.self)`), we are saying the elements are JSObject, but we still need to cast them when we go over the Array.

Reply:
Oh, I see. Your suggested change is functionally identical so I assumed it was just to be more concise. (I think a for loop with an inner guard is easier to understand but obviously that's subjective) I agree that bubbling the more exact types through the code is better but I was trying to avoid too many changes, especially since I knew the plugins were being split out and I didn't expect this change to get merged quickly. I will update the code to avoid casting types up to a more generic one.